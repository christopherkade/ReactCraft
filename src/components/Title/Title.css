.rc-title {
  color: #FFC700;
  font-weight: 300;
  -webkit-text-stroke-width: 0.2px;
  -webkit-text-stroke-color: black;
  text-shadow: 1.2px 1.5px black;
  margin: 0;
}
