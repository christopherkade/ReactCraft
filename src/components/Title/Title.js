import React from "react"
import PropTypes from 'prop-types';
import styled from "styled-components"

const StyledTitle = styled.h1`
  color: #FFC700;
  font-weight: 300;
  -webkit-text-stroke-width: 0.2px;
  -webkit-text-stroke-color: black;
  text-shadow: 1.2px 1.5px black;
  margin: 0;
`

const Title = ({ children }) => {
  return (
    <StyledTitle>
      {children}
    </StyledTitle>
  )
}

Title.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Title.defaultProps = {
  children: '',
};

export default Title
