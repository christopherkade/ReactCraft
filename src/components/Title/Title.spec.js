import React from 'react';
import Title from './Title';
import { render } from 'react-testing-library';

describe('<Title />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Title />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('children prop', () => {
    it('should display Eivin as title children', () => {
      const { container } = render(<Title>Eivin</Title>)
      expect(container.querySelector('h1').textContent).toBe('Eivin')
    })
  })
})
