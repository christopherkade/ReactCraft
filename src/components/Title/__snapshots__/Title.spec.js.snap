// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Snapshots should match snapshot 1`] = `
<div>
  <h1
    class="sc-bdVaJa cUfhWq"
  />
</div>
`;
